### PR TITLE
[ROCm] Optimize ROCm CI to reduce time

### PR DESCRIPTION
**Pull request by PeixuanZuo**
This PR mainly optimize ROCm CI test to reduce time and CPU utilization.

- use smaller batch size on strided_batched_gemm/batched_gemm test
- disable cpu training test
- fix test_e2e_padding_elimination Occasional failures on ROCm.

### CODE REVIEW

File: orttraining/orttraining/test/python/orttraining_ortmodule_tests.py [hunk 1]
```diff
@@ -178,15 +179,17 @@ def main():
 
     run_ortmodule_poc_net(cwd, log, no_cuda=False, data_dir=args.mnist)
 
-    run_ortmodule_poc_net(cwd, log, no_cuda=True, data_dir=args.mnist)
+    if os.getenv("ORTMODULE_ROCM_TEST", "0") == "0":
```
Comment (cloudhan) [inline orttraining/orttraining/test/python/orttraining_ortmodule_tests.py:182]: What is the purpose of this?

Comment (PeixuanZuo) [reply orttraining/orttraining/test/python/orttraining_ortmodule_tests.py:182]: disable cpu training test. It will take nearly 15 mintues.

Comment (cloudhan) [reply orttraining/orttraining/test/python/orttraining_ortmodule_tests.py:182]: This is not specific to rocm, can we rename it to `ORTMODULE_DISABLE_CPU_TEST` or something the like?

File: onnxruntime/python/tools/kernel_explorer/kernels/batched_gemm_test.py [hunk 2]
```diff
@@ -12,6 +13,8 @@
 import pytest
 from utils import get_gemm_basic_sizes, get_gemm_bert_sizes, get_gemm_bound, transab_to_suffix
 
+max_batch_size = int(os.environ.get("BATCHED_GEMM_MAX_BATCH_SIZE", 64))
```
[the remark this answers is not in the document]
Comment (PeixuanZuo) [reply onnxruntime/python/tools/kernel_explorer/kernels/batched_gemm_test.py]: Done.

File: orttraining/orttraining/test/python/orttraining_ortmodule_tests.py (hunk 1 above)
Comment (PeixuanZuo) [reply orttraining/orttraining/test/python/orttraining_ortmodule_tests.py]: Done.

File: tools/ci_build/github/pai/pai_huggingface_bert_large_test.sh [hunk 3]
```diff
@@ -3,7 +3,7 @@
 set -ex
 
 rocm_version=$1
-mi200_gpus=$(rocm-smi --showproductname | grep -c "MI250")
+mi200_gpus=$(rocm-smi --showproductname | grep "MI250" | wc -l)
```
Comment (cloudhan) [inline tools/ci_build/github/pai/pai_huggingface_bert_large_test.sh:6]: Please address the lint suggest here.

Comment (PeixuanZuo) [reply tools/ci_build/github/pai/pai_huggingface_bert_large_test.sh]: Done.

**Review by cloudhan — APPROVED** [review]

File: tools/ci_build/github/azure-pipelines/orttraining-pai-ci-pipeline.yml [hunk 4]
```diff
@@ -137,7 +137,7 @@ jobs:
     inputs:
       script: |-
         echo "Select agent: $(Agent.Name), GPU: $HIP_VISIBLE_DEVICES, render: $DRIVER_RENDER"
-        bash $(Build.SourcesDirectory)/tools/ci_build/github/pai/pai_get_thread.sh $(Agent.Name) $HIP_VISIBLE_DEVICES
+        bash $(Build.SourcesDirectory)/tools/ci_build/github/pai/pai_clean_device.sh $(Agent.Name) $HIP_VISIBLE_DEVICES
```
Comment (mszhanyi) [inline tools/ci_build/github/azure-pipelines/orttraining-pai-ci-pipeline.yml:140]: If the pai_clean_device.sh failed, should the pipeline continue?

Comment (PeixuanZuo) [reply tools/ci_build/github/azure-pipelines/orttraining-pai-ci-pipeline.yml:140]: we need stop pipeline if clean device step failed.

Comment (PeixuanZuo) [reply tools/ci_build/github/azure-pipelines/orttraining-pai-ci-pipeline.yml:140]: Done.

**Review by mszhanyi — APPROVED** [review]

File: tools/ci_build/github/azure-pipelines/orttraining-pai-ci-pipeline.yml [hunk 5]
```diff
@@ -240,11 +241,21 @@ jobs:
...
+              export ORTMODULE_DISABLE_CPU_TRAINING_TEST=1; \
+              export ORTMODULE_ROCM_TEST=1; \
               python orttraining_ortmodule_tests.py \
                 --mnist /home/onnxruntimedev/mnist \
                 --bert_data /home/onnxruntimedev/bert_data/hf_data/glue_data/CoLA/original/raw"
       workingDirectory: $(Build.SourcesDirectory)
     displayName: 'Run orttraining_ortmodule_tests.py'
     condition: succeededOrFailed()
 
+  - task: CmdLine@2
```
[the remark this answers is not in the document]
Comment (PeixuanZuo) [reply tools/ci_build/github/azure-pipelines/orttraining-pai-ci-pipeline.yml:253]: I will change this in the next PR.

**Pull request merged**